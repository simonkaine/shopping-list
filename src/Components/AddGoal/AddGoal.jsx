import React from 'react'
import { useState } from 'react';

export default function AddGoal({addGoal}) {
    const [goal, setGoal] = useState('');

    const submitHandle = (e) => {
        e.preventDefault();
        setGoal('');
        addGoal(goal);
    };


    return (
        <div>
            <form onSubmit={submitHandle}>
                <input 
                    type='text' 
                    placeholder='Enter a goal!'
                    value={goal} onChange={(e) => setGoal(e.target.value)} />

                <button type='submit'>
                    Add GOAL!
                </button>
            </form>
        </div>
    )
}
