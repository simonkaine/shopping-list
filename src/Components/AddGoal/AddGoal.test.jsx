import React from 'react';
import { screen, render } from '@testing-library/react';
import AddGoal from './AddGoal';

it('Should test for input to add a goal', async () => {
    const {container} = render(
        <AddGoal />
    )
    const labelInput = screen.getByLabelText(/Enter/i)
    expect(labelInput).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})
