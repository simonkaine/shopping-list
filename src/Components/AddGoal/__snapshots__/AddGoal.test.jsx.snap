// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test for input to add a goal 1`] = `
<div>
  <div>
    <form>
      <label
        for="addGoal"
      >
        Enter: 
      </label>
      <input
        id="addGoal"
        placeholder="Enter a goal!"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add GOAL!
      </button>
    </form>
  </div>
</div>
`;
